src/prop-types/prop-type-from-metadata.ts

```diff
@@ -47,4 +57,8 @@ const propTypeClassByType: {
     ifTypeMatches: /^(undefined \| )?{ (\w+: .+; )+}$/, // { key1: type1; key2: type2; }
     thenTypeClass: FixedObjectType,
   },
+  {
+    ifTypeMatches: /^(undefined \| )?(.+ \| )+[^\|]+$/, // supportedType1 | supportedType2
```

Comment:
Would be good to go through this one together to make sure I'm on the same page.

Reply:
Happy to take you through the code, but I'm going to merge this now so I can build on top of it.